PSCompatibilityCollector/Microsoft.PowerShell.CrossCompatibility/Commands/ConvertJsonCommands.cs

```diff
@@ -0,0 +1,116 @@
+        /// </summary>
+        [Parameter]
+        [Alias("Compress")]
+        public SwitchParameter NoWhitespace { get; set; }
+
+        protected override void BeginProcessing()
+        {
+            _serializer = JsonProfileSerializer.Create(NoWhitespace ? Formatting.None : Formatting.Indented);
+        }
+
+        protected override void ProcessRecord()
+        {
+            foreach (object obj in Item)
+            {
+                WriteObject(_serializer.Serialize(obj));
+            }
+            return;
+        }
+    }
+
+    /// <summary>
+    /// Class defining the ConvertFrom-PSCompatibilityJson JSON.
+    /// Turns given JSON into .NET compatibility profile objects.
+    /// </summary>
+    [Cmdlet(VerbsData.ConvertFrom, CommandUtilities.MODULE_PREFIX + "Json", DefaultParameterSetName = "JsonSource")]
+    public class ConvertFromPSCompatibilityJsonCommand : PSCmdlet
+    {
+        private JsonProfileSerializer _serializer;
+
+        /// <summary>
+        /// The source of the JSON to convert from. Can be a JSON string, a FileInfo or a TextReader object.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true, ParameterSetName = "JsonSource")]
+        public object[] JsonSource { get; set; }
+
+        /// <summary>
+        /// The path of a file to convert JSON from.
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = "Path")]
+        [ValidateNotNullOrEmpty()]
+        public string[] Path { get; set; }
+
+        public ConvertFromPSCompatibilityJsonCommand()
+        {
+            _serializer = JsonProfileSerializer.Create();
+        }
+
+        protected override void ProcessRecord()
+        {
+            if (Path != null)
+            {
+                foreach (string filePath in Path)
+                {
+                    string absolutePath = this.GetNormalizedAbsolutePath(filePath);
+                    WriteObject(_serializer.DeserializeFromFile(absolutePath));
+                }
+                return;
+            }
+
+            if (JsonSource != null)
+            {
+                foreach (object jsonSourceItem in JsonSource)
+                {
+                    switch (jsonSourceItem)
+                    {
+                        case string jsonString:
+                            WriteObject(_serializer.Deserialize(jsonString));
+                            return;
+
+                        case FileInfo jsonFile:
+                            WriteObject(_serializer.Deserialize(jsonFile));
+                            return;
+
+                        case TextReader jsonReader:
+                            WriteObject(_serializer.Deserialize(jsonReader));
+                            return;
+
+                        default:
+                            throw new ArgumentException($"Unsupported type for {nameof(JsonSource)} parameter. Should be a string, FileInfo or TextReader object.");
```

Comment:
I think this should probably be a `WriteError` because it takes a collection of objects so one bad one should not cause the pipeline to abort. Also, if it should be terminating, then you probably want `ThrowTerminatingError`

Reply:
Ah ok! Is it possible to throw multiple terminating errors?